src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -239,6 +243,50 @@ private static void createBranches(LfNetwork lfNetwork, LoadingContext loadingCo
                 legNumber++;
             }
         }
+
+        // Merge the discrete voltage control in each non-impedant connected set
+        Graph<LfBus, LfBranch> nonImpedantSubGraph = getNonImpedantSubGraph(lfNetwork);
+        List<Set<LfBus>> connectedSets = new ConnectivityInspector<>(nonImpedantSubGraph).connectedSets();
+        connectedSets.forEach(LfNetworkLoaderImpl::mergeConflictuousDiscreteVoltageControls);
+    }
+
+    public static Graph<LfBus, LfBranch> getNonImpedantSubGraph(LfNetwork network) {
+        List<LfBranch> nonImpedantBranches = network.getBranches().stream()
+            .filter(LfNetworkLoaderImpl::isNonImpedantBranch)
+            .filter(b -> b.getBus1() != null && b.getBus2() != null)
+            .collect(Collectors.toList());
+
+        Graph<LfBus, LfBranch> nonImpedantSubGraph = new Pseudograph<>(LfBranch.class);
+        for (LfBranch branch : nonImpedantBranches) {
+            nonImpedantSubGraph.addVertex(branch.getBus1());
+            nonImpedantSubGraph.addVertex(branch.getBus2());
+            nonImpedantSubGraph.addEdge(branch.getBus1(), branch.getBus2(), branch);
+        }
+
+        return  nonImpedantSubGraph;
+    }
+
+    public static boolean isNonImpedantBranch(LfBranch branch) {
+        PiModel piModel = branch.getPiModel();
+        return piModel.getZ() < DcEquationSystem.LOW_IMPEDANCE_THRESHOLD;
+    }
+
+    private static void mergeConflictuousDiscreteVoltageControls(Set<LfBus> nonImpedantConnectedSet) {
```

Comment:
Great! We have to increase again this function that will become more a `fixDiscreteVoltageControls` indeed as:
- Here we merge discrete voltage control if we have more than one bus controlled by RTC in a connected set of non impedant branches ;
- We also have to delete all the discrete voltage controls if we find in the connected set a bus controlled by a generator or a SVC (`isVoltageControlled`). In a second time (this year but not now), we will have to deal with mixed shared controls.